text/3850-autogenerated-attr.md

```diff
@@ -0,0 +1,121 @@
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This has no direct effect on how Rust is parsed. Instead, each tool will dictate how to understand the attribute based on what the tool is/does. Effects that commonly used tools would need to worry about were specified in the [motivation section](motivation). Most tools would need to worry about parsing the `by` attribute and whichever others affect the tool itself, and then use the `by` attribute for better error messages. For example, instead of:
+
+```plain
+error: expected one of `(`, `[`, or `{`, found `<eof>`  
+--> foobar.rs:2:1
+|  
+2  | not rust
+|  ^  expected one of `(`, `[`, or `{`  
+  
+error: aborting due to 1 previous error
+```
+
+This message could be output:
+
+```plain
+error: expected one of `(`, `[`, or `{`, found `<eof>`  
+--> foobar.rs:2:1 - generated by example tool (is there a bug in this tool?)
+error: aborting due to 1 previous error
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There aren't many drawbacks - if a tool doesn't understand the attribute then it should just skip over it. One problem that could arise is tools having their own parameters for the attribute that aren't specified, and it's unclear how to work around that.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Unlike many other languages with a similar feature, we have attributes which can be used for easy scoping like this. Other possible designs commonly used in other languages are a simple comment, such as Go's regex matching at the start of a file:
+```regex
+^// Code generated .* DO NOT EDIT.$
+```
+or C#'s:
+```csharp
+// <auto-generated>
+// (arbitrary text, usually a long description of the file)
+// </auto-generated>
+```
+However, these both have the same problem: All that tools can do is check if a file is autogenerated, and nothing else. The design being suggested in this RFC allows tools to configure themselves to work better with autogenerated files and even see which tool made them. Another problem with this comment approach is it only supports files, not arbitrary items.
+
+This could in theory be done in a tool, however it works much better as a language feature for consistency's sake. Also, if it were done in a tool, where would it be placed? It wouldn't fit in any of the current ones given it's general and wouldn't be worth creating a new one for this. That's a bit of a nitpick however.
+
+This also improves code readability, allowing users and tools alike to easily see which tool created the generated code and even roughly how it was created (through `regen_hint`).
+
+Some things for individual tools do already exist such as `#[rustfmt::skip]`, however these of course apply only to individual tools meaning that a more general attribute would work well for this scenario.
+
+# Prior art
```

Comment:
A good addition to this section would be a summary of what *effects* these “generated” markers have on tools in other languages.